src/quarkusProjectListener.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2019 Red Hat, Inc. and others.
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { commands, workspace } from 'vscode';
+import { checkQuarkusProjectExistsWorkspace } from './utils/workspaceUtils';
+
+export function createQuarkusProjectListener() {
+  return workspace.onDidChangeWorkspaceFolders(async () => {
+    await new Promise((res => setTimeout(res, 100)));
```

Comment:
This was done because without it, the results from the `java/projectTypes` isn't accurate.

For example, if I have 3 quarkus projects in my workspace and I remove one, the `java/projectTypes`  command gets called, and it returns information that indicates that I still have 3 quarkus projects in my workspace, when I only have 2.